Code/GraphMol/MarvinParse/MarvinParser.cpp

```diff
@@ -172,6 +172,7 @@ class MarvinCMLReader {
           query->setProp(common_properties::molFileAlias, marvinAtom->mrvAlias);
         }
         query->setQuery(makeAtomNullQuery());
+        query->setIsotope(marvinAtom->rgroupRef);
```

Comment:
There's actually one additional piece of info you should set in order to be sure that the R label is handled everywhere, the `_MolFileRLabel` property:
```suggestion
        query->setIsotope(marvinAtom->rgroupRef);
        query->setProp(common_properties::_MolFileRLabel, static_cast<unsigned int>(marvinAtom->rgroupRef))
```
I agree that this is all a mess, but I'm doubtful that there's any way to rationalize it without breaking other people's code.